src/Providers/Http/HttpTransporterFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Http;
+
+use Http\Discovery\Psr17FactoryDiscovery;
+use Http\Discovery\Psr18ClientDiscovery;
+use WordPress\AiClient\Providers\Http\Contracts\HttpTransporterInterface;
+
+/**
+ * Factory for creating HTTP transporters.
+ *
+ * Uses HTTPlug's Discovery component to automatically find
+ * available HTTP clients and factories.
+ *
+ * @since n.e.x.t
+ */
+class HttpTransporterFactory
```

Comment:
We haven't used the factory pattern anywhere else. I tend to agree it makes sense here, but just flagging it, since it may beg the question where else (if anywhere) we should use it, and what is our criteria for using it vs just manually instantiating a class? What do you think?

Reply:
I think let's leave it for now. We can revise it later if we find a better place.